code/game/atoms_movable.dm

```diff
@@ -403,6 +403,32 @@
 
 	set_throwing(TRUE)
 	src.thrower = thrower
+
+	move_at(target,range,speed,TRUE)
+
+	//done throwing, either because it hit something or it finished moving
+	if(isobj(src) && throwing)
+		throw_impact(get_turf(src), speed)
+	if(loc)
+		set_throwing(FALSE)
+		thrower = null
+		throw_source = null
+
+/atom/movable/proc/fly_at(atom/target, range, speed)
+	set waitfor = FALSE
+	var/init_layer = layer
+	layer = FLY_LAYER
+	if(!target || !src)
+		return FALSE
+
+	set_flying(TRUE)
+	move_at(target, range, speed, FALSE)
+	if(loc)
+		set_flying(FALSE)
+		throw_source = null
+		layer = init_layer
```

Comment:
this should be throw_at
If you want the flying stuff just add a set_flying before and after

Reply:
That would also mean undoing the changes to the throwing procs